### PR TITLE
Internal: Fix form submission labels [ED-23179]

### PR DESCRIPTION
**Pull request by marcin-el**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### ✨ PR Description
Purpose: Fix form field label resolution priority to follow proper accessibility hierarchy, ensuring aria-label and semantic HTML labels take precedence over placeholder text in form submissions.

Main changes:
- Reordered label detection logic to prioritize aria-label, then for-label, then parent label, with placeholder as fallback
- Added early return for parent label text to prevent fallthrough to placeholder when parent label exists
- Added comprehensive test suite covering all four label resolution priority scenarios with edge cases

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using.
💡 **Tip:** You can customize your AI Description using **Guidelines** [Learn how](https://docs.gitstream.cm/automation-actions/#describe-changes)</sub>

### CODE REVIEW

File: modules/atomic-widgets/assets/js/frontend/handlers.js [hunk 1]
```diff
@@ -181,7 +176,12 @@ function getAtomicFormFieldLabel( field, form ) {
 	}
 
 	const parentLabelText = field.closest( 'label' )?.textContent?.trim();
-	return parentLabelText || '';
+	if ( parentLabelText ) {
+		return parentLabelText;
+	}
+
+	const placeholder = field.getAttribute( 'placeholder' );
+	return placeholder || '';
```
Comment (murzin-nikita) [inline modules/atomic-widgets/assets/js/frontend/handlers.js:185]:
```suggestion
	const placeholder = field.getAttribute( 'placeholder' );
	
	return placeholder || '';
```

**Review by murzin-nikita — DISMISSED** [review]

**Review by murzin-nikita — APPROVED** [review]

**Review by RobiFerentz — APPROVED** [review]
